flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/source/config/MongoReadOptions.java

```diff
@@ -0,0 +1,224 @@
+        private int cursorBatchSize = SCAN_CURSOR_BATCH_SIZE.defaultValue();
+        private boolean noCursorTimeout = SCAN_CURSOR_NO_TIMEOUT.defaultValue();
+        private PartitionStrategy partitionStrategy = SCAN_PARTITION_STRATEGY.defaultValue();
+        private MemorySize partitionSize = SCAN_PARTITION_SIZE.defaultValue();
+        private int samplesPerPartition = SCAN_PARTITION_SAMPLES.defaultValue();
+
+        /**
+         * Sets the number of documents should be fetched per round-trip when reading.
+         *
+         * @param fetchSize the number of documents should be fetched per round-trip when reading.
+         * @return this builder
+         */
+        public MongoReadOptionsBuilder setFetchSize(int fetchSize) {
+            checkArgument(fetchSize > 0, "The fetch size must be larger than 0.");
+            this.fetchSize = fetchSize;
+            return this;
+        }
+
+        /**
+         * Sets the batch size of MongoDB find cursor.
+         *
+         * @param cursorBatchSize the max batch size of find cursor.
+         * @return this builder
+         */
+        public MongoReadOptionsBuilder setCursorBatchSize(int cursorBatchSize) {
+            checkArgument(
+                    cursorBatchSize >= 0,
+                    "The cursor batch size must be larger than or equal to 0.");
+            this.cursorBatchSize = cursorBatchSize;
+            return this;
+        }
+
+        /**
+         * Set this option to true to prevent cursor timeout (10 minutes).
+         *
+         * @param noCursorTimeout Set this option to true to prevent cursor timeout (10 minutes)
+         * @return this builder
+         */
+        public MongoReadOptionsBuilder setNoCursorTimeout(boolean noCursorTimeout) {
+            this.noCursorTimeout = noCursorTimeout;
+            return this;
+        }
+
+        /**
+         * Sets the partition strategy.
+         *
+         * @param partitionStrategy the strategy of a partition.
+         * @return this builder
+         */
+        public MongoReadOptionsBuilder setPartitionStrategy(PartitionStrategy partitionStrategy) {
+            checkNotNull(partitionStrategy, "The partition strategy must not be null.");
+            this.partitionStrategy = partitionStrategy;
+            return this;
+        }
+
+        /**
+         * Sets the partition size of MongoDB split.
+         *
+         * @param partitionSize the memory size of a partition.
+         * @return this builder
+         */
+        public MongoReadOptionsBuilder setPartitionSize(MemorySize partitionSize) {
+            checkNotNull(partitionSize, "The partition size must not be null");
+            checkArgument(
+                    partitionSize.getMebiBytes() >= 1,
+                    "The partition size must be larger than or equals to 1mb.");
+            this.partitionSize = partitionSize;
+            return this;
+        }
+
+        /**
+         * Sets the partition size of MongoDB split.
```

Comment:
javadocs are identical to setPartitionSize

Reply:
How about change it to the following?
```
Sets the partition memory size of MongoDB split. Split a MongoDB collection into multiple
partitions according to the partition memory size. We can read these partitions in parallel
to improve the reading speed.
```